java-manta-client-unshaded/src/main/java/com/joyent/manta/http/MantaConnectionFactory.java

```diff
@@ -86,10 +88,11 @@
     /**
      * Default HTTP headers to send to all requests to Manta.
      */
-    private static final Collection<? extends Header> HEADERS = Arrays.asList(
-            new BasicHeader(MantaHttpHeaders.ACCEPT_VERSION, "~1.0"),
-            new BasicHeader(HttpHeaders.ACCEPT, "application/json, */*")
-    );
+    public static final Collection<? extends Header> HEADERS =
+            Collections.unmodifiableList(
```

Comment:
Is this needed for this change?  It looks like a fine improvement but was unsure if it was demonstrating one of the "spooky"  issues with all this being user configurable.

Reply:
I remember punting on the decision between:

- document default headers and tell `MantaConnectionFactoryConfigurator` users to include these headers in their `HttpClientBuilder` configuration
- add these headers to whatever default headers are provided by the user. This might require inspecting the `HttpClientBuilder`'s internal state to determine if we've already added them in #311?
- add a `setDefaultHeaders` method to `MantaConnectionFactoryConfigurator` so users can tell us what headers they want to mix into the `HttpClientBuilder`
- something else?

Reply:
One thought is to change the `HEADERS` constant into an array of `org.apache.http.Header[]` objects and store it as a constant in `MantaHttpRequestFactory` or in `StandardHttpHelper` and add the headers to every `HttpHead`, `HttpGet`, etc. This would be a solution that would always make sure the right headers were there regardless of the `HttpClientBuilder` that was used.

Reply:
That suggestion led me to discover default headers are implemented as just another `HttpRequestInterceptor`, definitely going to use `MantaHttpRequestFactory` for that and remove our call to `HttpClientBuilder#setDefaultHeaders`. Thanks!